### ISSUE TITLE
this code causes wandb to consistently crash

### ISSUE THREAD

**Issue by lukas** [wandb/client#77]
It spawns a few threads - maybe thats why?

https://gist.github.com/lukas/2cfcc73fd4c7ad7cb6e807e29087d5b2

**Comment by shawnlewis** [comment]
It seems to get hung up when n_jobs>1. Maybe a missed process/thread issue. I can try to test switching all of our threads to processes. But the best solution might be to get all of the sync'ing stuff out of wandb.init() and into the "wandb run" process.

**Comment by shawnlewis** [comment]
More spelunking: I commented out everything in wandb.sync except upsert_run, and this hang still happened. Then I got rid of the call to upsert_run and it worked as normal. In upsert_run getting rid of self.client.execute causes normal behavior. Within the gql client, getting rid of execute's call to requests.post fixes the issue.

So there is some weirdness having to do with an interaction between requests and joblib's Parallel.

I think the right solution is for us to move as much as we can out of wandb.init() and into "wandb run" in the parent process. This'll be a big refactor but it fixes issues I'm having with rllab too.
